ycmd.el

```diff
@@ -260,6 +260,24 @@ use `ycmd-parse-buffer'."
   :group 'ycmd
   :type '(alist :key-type symbol :value-type (repeat string)))
 
+(defconst ycmd-semantic-completion-file-types
+  '("c"
+    "cpp"
+    "objc"
+    "objcpp"
+    "cs"
+    "python")
+  "A list of ycmd file type strings which support semantic completion.")
+
+(defun ycmd-semantic-file-types (mode)
+  "Find the ycmd file types for MODE which support semantic completion.
+
+Returns a possibly empty list of ycmd file type strings.  If this
+is empty, then ycmd doesn't support semantic completion (or
+diagnostics) for MODE."
+  (let ((file-types (assoc-default mode ycmd-file-type-map)))
+    (-intersection file-types ycmd-semantic-completion-file-types)))
```

Comment:
> As for memq, it seems less convenient than -intersection because I'd have to put it in a map or some loop since fileypes is a list containing – at least potentially – multiple strings.

Of course, you're right with `memq`.